test/specs/components/Popup/Popup-test.tsx

```diff
@@ -1,6 +1,5 @@
-import _ from 'lodash'
-import computePopupPlacement from 'src/components/Popup/positioningHelper'
-import { Position, Alignment } from 'src/components/Popup/Popup'
```

Comment:
Crazy, these are not even valid exports 😕 I wonder how they were even compiling...

Reply:
Crazier thing I just found out today that explains this:

**We are not including `test/` folder in our typescript compiler options so nothing under tests is compiled by TypeScript.**

I did some research and tracked it down to `build/gulp/tasks/dist.ts`. The `build` task relies on the `paths` object defined in `config.ts` and there is nothing related to `test`.

I made the necessary changes to enable typescript compilation for `test` folder and we have 50+ errors, among them:

test/specs/components/Popup/Popup-test.tsx(3,10): error TS2305: Module '"/Users/alex/work/react/src/components/Popup/Popup"' has no exported member 'Position'.
test/specs/components/Popup/Popup-test.tsx(3,20): error TS2305: Module '"/Users/alex/work/react/src/components/Popup/Popup"' has no exported member 'Alignment'.

I talked to @smykhailov about this as well and he suggested that:
- we should not rely on all these scripts and tools (such as `gulp` / `grunt`) for our build because:
  - it can introduce lots of errors as this one;
  - it duplicates a lot of work;
  - we end up creating build scripts everywhere that are hard to understand and maintain.
- we should leverage more the `webpack` functionality for setting up our build. 

Do we have any issue related to our build system approach and tooling?

What is your take on this?
@levithomason @kuzhelov @miroslavstastny @mnajdova @alinais @smykhailov